diff --git a/lib/libdivide/libdivide.h b/lib/libdivide/libdivide.h
--- a/lib/libdivide/libdivide.h
+++ b/lib/libdivide/libdivide.h
@@ -2093,7 +2093,7 @@
         }
 
         bool operator==(const divider<T, ALGO> &other) const {
-            return div.denom.magic == other.denom.magic && div.denom.more == other.denom.more;
+            return div.denom.magic == other.div.denom.magic && div.denom.more == other.div.denom.more;
         }
 
         bool operator!=(const divider<T, ALGO> &other) const {
